sklearn/utils/estimator_checks.py

```diff
@@ -1915,6 +1917,42 @@ def transform(self, X):
                     shallow_params.items()))
 
 
+@ignore_warnings(category=(DeprecationWarning, FutureWarning))
+def check_set_params(name, estimator_orig):
+    # Check that get_params() returns the same thing
+    # before and after set_params() with some fuzz
+    estimator = clone(estimator_orig)
+
+    params = estimator.get_params()
+    msg = ("get_params result does not match what was passed to set_params")
+
+    estimator.set_params(**params)
+    new_params = estimator.get_params()
+    assert_equal(params.keys(), new_params.keys(), msg)
+    for k, v in new_params.items():
+        assert_is(params[k], v, msg)
+
+    # some fuzz values
+    test_values = [-np.inf, np.inf, None]
+
+    for param_name in params.keys():
+        default_value = params[param_name]
+        for value in test_values:
+            params[param_name] = value
+            try:
+                estimator.set_params(**params)
+            except Exception:
+                # triggered some parameter validation
```

Comment:
In the ideal world, we might not permit this validation outside of fit either. I think we should consider raising a warning of some kind where this case is over-used.

Reply:
I see your point but when is the developer expected to see the warning? As part of the (massive) output of the test suite?

Reply:
the output of the test-suite should not be massive. There are a lot of issues in master because we were not careful about handling warnings.

Reply:
+1 for raising a warning (or even error?) here. Did this error for all_estimators? where? Usually each new test in ``estimator_checks`` requires fixes to the codebase. This one here is probably the one that should have been fixed. Not having any errors on the codebase is suspicious to me.

Reply:
For a second I thought running this on a deep estimator would fail, but I guess we avoid this by doing ``deep=False``. It would be really nice to run ``estimator_checks`` on some deep estimators though (I tried and ran into a whole host of problems)